Loop/Managers/AlertMuter.swift

```diff
@@ -108,13 +103,20 @@ public class AlertMuter: ObservableObject {
         self.init(configuration: Configuration(startTime: startTime, duration: duration))
     }
 
-    func check(_ now: Date = Date()) {
-        // this is enabled by user action, so only need to check if the duration has elapsed and remove the startTime
-        guard let startTime = configuration.startTime,
-              startTime.addingTimeInterval(configuration.duration) < now
-        else { return }
+    private func updateMutePeriodEndingWatcher(_ now: Date = Date()) {
+        mutePeriodEndingTimer?.invalidate()
 
-        configuration.startTime = nil
+        guard let mutingEndTime = configuration.mutingEndTime else { return }
+
+        guard mutingEndTime > now else {
+            configuration.startTime = nil
+            return
+        }
+
+        let timeInterval = mutingEndTime.timeIntervalSince(now)
+        mutePeriodEndingTimer = Timer.scheduledTimer(withTimeInterval: timeInterval, repeats: false) { [weak self] _ in
```

Comment:
Sorry, I meant that I was thinking it would be more appropriate for the *UI* components to own the timers. If there is nothing onscreen that needs to be updated, there is no need for timers.  The AlertMuter itself is able to return a correct value for isMuting without timers. I won't block on this, if you prefer this arrangement, as it should work fine.

Reply:
I thought this would be a simpler solution. The UI may not know it needs a timer to track this since configuration is published.